builtin/provisioners/remote-exec/resource_provisioner.go

```diff
@@ -272,7 +272,7 @@ func retryFunc(ctx context.Context, timeout time.Duration, f func() error) error
 			}
 
 			// Try the function call
-			err := f()
+			err := backoff.Retry(f, backoff.NewExponentialBackOff())
```

Comment:
We should probably update the backoff package and use the `WithContext` function too, so that the Retry would get canceled with the context, otherwise it will keep running in the background.